File: src/protocol/UserOutOfHomeCommand.cpp
#include "UserOutOfHomeCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

UserOutOfHomeCommand::UserOutOfHomeCommand()
        : Command(10, CommandType::USER_OUT_OF_HOME)
{}

UserOutOfHomeCommand::~UserOutOfHomeCommand()
{}

vector<uint8_t> UserOutOfHomeCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

string UserOutOfHomeCommand::getName()
{
    return string("UserOutOfHomeCommand");
}

void UserOutOfHomeCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void UserOutOfHomeCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/UserOutOfHomeCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public final class UserOutOfHomeCommand extends Command
	{
		public UserOutOfHomeCommand()
		{
			super(10, CommandType.USER_OUT_OF_HOME);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("UserOutOfHomeCommand");
		}

		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}